### PR TITLE
Update muted words dialog with `expiresAt` and `actorTarget`

### PR DESCRIPTION
**Pull request by estrattonbailey**
### Relies on https://github.com/bluesky-social/atproto/pull/2276

Adds support for timed mute words, as well as targeting only those users whom you don't follow. Also integrates new handlers into the `TagMenu` (when clicking or long-pressing on a tag).

This PR also adds a new hook `useFormatDistance`, with the intent that we can add more in the pattern of `use<method>` where `method` is a `date-fns` method that accepts localization. Right now, all our app languages are supported with the exception of Gaelic. In the interest of time and stability (I'd like to use a library for this) I think this is an OK trade-off. We can help facilitate a `ga` translation in the near future.

image

i18n works:
CleanShot 2024-07-19 at 14 24 27@2x

### CODE REVIEW

**Comment by surfdude29** [comment]
While you're working on updating the muted words dialog, might it be possible to squeeze in a fix for #3311? 🙏

This issue affects the web interface in Catalan, Finnish, French, German, Irish, Portuguese and Ukrainian.

File: src/components/dialogs/MutedWords.tsx [hunk 1]
```diff
@@ -53,45 +56,68 @@ function MutedWordsInner() {
...
     } catch (e: any) {
       logger.error(`Failed to save muted word`, {message: e.message})
       setError(e.message)
     }
-  }, [_, field, options, addMutedWord, setField])
+  }, [_, field, targets, addMutedWord, setField, durations, excludeFollowing])
 
   return (
     <Dialog.ScrollableInner label={_(msg`Manage your muted words and tags`)}>
-      <View onTouchStart={Keyboard.dismiss}>
```
Comment (haileyok) [inline src/components/dialogs/MutedWords.tsx:83]: tested on ios/android to make sure this is okay to remove, seems good. think using the discord fork of bottomsheet made this unnecessary.

File: src/components/dialogs/MutedWords.tsx [hunk 2]
```diff
@@ -107,30 +133,130 @@ function MutedWordsInner() {
             }}
             onSubmitEditing={submit}
           />
+        </View>
 
+        <View style={[a.pb_xl, a.gap_sm]}>
           <Toggle.Group
-            label={_(msg`Toggle between muted word options.`)}
+            label={_(msg`Select how long this mute word should apply for.`)}
```
Comment (haileyok) [inline src/components/dialogs/MutedWords.tsx:140]:
```suggestion
            label={_(msg`Select how long to mute this word for.`)}
```

File: src/components/dialogs/MutedWords.tsx [hunk 3]
```diff
@@ -289,53 +438,77 @@ function MutedWordRow({
...
-          {word.targets.map(target => (
-            <View
-              key={target}
-              style={[a.py_xs, a.px_sm, a.rounded_sm, t.atoms.bg_contrast_100]}>
-              <Text
-                style={[a.text_xs, a.font_bold, t.atoms.text_contrast_medium]}>
-                {target === 'content' ? _(msg`text`) : _(msg`tag`)}
+              <Text style={[a.font_normal, t.atoms.text_contrast_medium]}>
+                {' '}
+                <Trans>in</Trans>{' '}
```
Comment (haileyok) [inline src/components/dialogs/MutedWords.tsx:465]: Can we move `<Trans>` to wrap the entire root `<Text>` here starting on L451?

**Review by haileyok — COMMENTED** [review]
Cooooool

**Comment by estrattonbailey** [comment]
Contrived example, but text can now handle wrapping for languages with longer transations.

CleanShot 2024-07-23 at 15 17 22@2x

**Comment by estrattonbailey** [comment]
Not sure how ya saw those borked views, might be without proper types/validation if you didn't have the linked API SDK built and installed. Was able to fix the label alignment though, wonder when that got bumped.

**Comment by surfdude29** [comment]
> Contrived example, but text can now handle wrapping for languages with longer transations.

Excellent, thanks! :)

**Review by pfrazee — APPROVED** [review]
Tested on all platforms, didn't test new logic correctness but the tests in the api seemed solid

LGTM!

**Pull request merged**